### PR TITLE
fix: unescape tabs before writing to pom.xml

### PR DESCRIPTION
**Pull request by cuixq**
https://github.com/google/osv-scanner/issues/1184

When encoding tokens to pom.xml, tabs are escaped and this is not what we want.
In this PR, before writing to pom.xml, we replace all escaped tabs with unescaped characters.

### CODE REVIEW

**Review by oliverchang — COMMENTED** [review]
nice! just have some questions:
- will this result in unrelated diffs in the result? 
- is there a bug in the Go encoding/xml package, or some more context you can give on why this occurs?

**Comment by cuixq** [comment]
> nice! just have some questions:
> 
> * will this result in unrelated diffs in the result?

Yes, but I think this is definitely better than having `&#x9;` in the result.

> * is there a bug in the Go encoding/xml package, or some more context you can give on why this occurs?

I believe this is due to `EncodeToken` escaping the text: https://github.com/golang/go/blob/master/src/encoding/xml/marshal.go#L222

Also from https://github.com/golang/go/issues/47464, I don't think Go refers this as a bug.

**Comment by oliverchang** [comment]
> > nice! just have some questions:
> > 
> > * will this result in unrelated diffs in the result?
> 
> Yes, but I think this is definitely better than having `&#x9;` in the result.

This is a bit of a hack, but is it possible to replace this with a sentinel value. e.g. "REPLACEWITHTAB", and then replace this ourselves when we're about to write it to the output?

> 
> > * is there a bug in the Go encoding/xml package, or some more context you can give on why this occurs?
> 
> I believe this is due to `EncodeToken` escaping the text: https://github.com/golang/go/blob/master/src/encoding/xml/marshal.go#L222
> 
> Also from [golang/go#47464](https://github.com/golang/go/issues/47464), I don't think Go refers this as a bug.

Got it, so it's correct but non-ideal behaviour.

**Comment by cuixq** [comment]
I think replacing with some sentinel value may not work since this will make the pom.xml no longer valid. However this hints me that maybe we could let the texts be escaped during writing and replace all escaped `&#x9;` by `\t` at the end. I will investigate the possibility.

**Comment by oliverchang** [comment]
> I think replacing with some sentinel value may not work since this will make the pom.xml no longer valid. However this hints me that maybe we could let the texts be escaped during writing and replace all escaped `&#x9;` by `\t` at the end. I will investigate the possibility.

yeah, great point. we can just unescape if we are able to hook into that part of the output.

File: internal/resolution/manifest/__snapshots__/maven_test.snap [hunk 1]
```diff
@@ -26,7 +170,9 @@
     <maven.compiler.source>1.7</maven.compiler.source>
     <maven.compiler.target>1.7</maven.compiler.target>
     <junit.version>4.13.2</junit.version>
-    <zeppelin.daemon.package.base>../bin</zeppelin.daemon.package.base>
+&#x9;  <zeppelin.daemon.package.base>
+&#x9;    ../bin
+&#x9;  </zeppelin.daemon.package.base>
```
Comment (michaelkedar) [inline internal/resolution/manifest/__snapshots__/maven_test.snap:175]: There seems to be escaped tabs in this test snapshot?

Comment (cuixq) [reply internal/resolution/manifest/__snapshots__/maven_test.snap:175]: this is expected - this is result from `write()` and then we unescape all the tabs before writing to the actual file in `Write()`

**Review by oliverchang — APPROVED** [review]

**Review by michaelkedar — APPROVED** [review]

**Pull request merged**